### PR TITLE
[zuul-core] Add counter for conn. close on idle state

### PR DESCRIPTION
**Pull request by argha-c**

### CODE REVIEW

**Comment by artgon** [comment]
This seems like a duplicate of this feature: https://github.com/Netflix/zuul/blob/master/zuul-core/src/main/java/com/netflix/netty/common/metrics/ServerChannelMetrics.java#L107

**Comment by argha-c** [comment]
> This seems like a duplicate of this feature: https://github.com/Netflix/zuul/blob/master/zuul-core/src/main/java/com/netflix/netty/common/metrics/ServerChannelMetrics.java#L107

Hmm, indeed. I can port the test, though it'd be nice if channel handlers encapsulated their own metrics imho.

File: zuul-core/src/test/java/com/netflix/netty/common/CloseOnIdleStateHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    public void incrementCounterOnIdleStateEvent() {
+        final EmbeddedChannel channel = new EmbeddedChannel();
+        channel.pipeline().addLast(new DummyChannelHandler());
+        channel.pipeline().addLast(new CloseOnIdleStateHandler(registry, listener));
+
+        channel.pipeline().context(DummyChannelHandler.class).fireUserEventTriggered(ALL_IDLE_STATE_EVENT);
+
+        verify(counter, times(1)).increment();
+    }
+}H
```
Comment (argha-c) [inline zuul-core/src/test/java/com/netflix/netty/common/CloseOnIdleStateHandlerTest.java:59]: Wut!

File: zuul-core/src/test/java/com/netflix/netty/common/CloseOnIdleStateHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CloseOnIdleStateHandlerTest {
+
+    @Mock
+    private Registry registry;
```
Comment (carl-mastrangelo) [inline zuul-core/src/test/java/com/netflix/netty/common/CloseOnIdleStateHandlerTest.java:37]: Consider using a default registry and checking the metric value at the end of the test

Comment (argha-c) [reply zuul-core/src/test/java/com/netflix/netty/common/CloseOnIdleStateHandlerTest.java:37]: yep, good call.

Comment (argha-c) [reply zuul-core/src/test/java/com/netflix/netty/common/CloseOnIdleStateHandlerTest.java:37]: actually, I don't see an API to lookup the counter for `DefaultRegistry`. 
(I'd have to inject the counter obj itself then, which isn't clean.)

**Pull request merged**